### DISCUSSION TITLE
How to clean up the source code path when using `gen_ref_pages.py`?

### DISCUSSION THREAD

**Discussion by Andrew-S-Rosen** [mkdocstrings/mkdocstrings#622] [Q&A] [answered]
I am using the sample [`gen_ref_pages.py`](https://mkdocstrings.github.io/recipes/#bind-pages-to-sections-themselves) script in the documentation. Everything looks great locally, but when I deploy my documentation via GitHub, I get the following:

![image](https://github.com/mkdocstrings/mkdocstrings/assets/8674072/c28367f5-6367-4c0c-804e-5429ae372595)

[Here](https://github.com/Quantum-Accelerators/quacc/blob/main/docs/gen_ref_pages.py) is my specific `gen_ref_pages.py` script, and [here](https://quantum-accelerators.github.io/quacc/reference/quacc/atoms/core.html) is a page highlighting the above image.

Do you have any recommendations about how to get the source code path to be only the path starting at my package name?

**Comment by pawamoy** [comment, marked as answer]
You could try [overriding templates](https://mkdocstrings.github.io/usage/theming/#templates) to use [`relative_package_filepath`](https://mkdocstrings.github.io/griffe/reference/griffe/dataclasses/#griffe.dataclasses.Object.relative_package_filepath) instead of [`relative_filepath`](https://mkdocstrings.github.io/griffe/reference/griffe/dataclasses/#griffe.dataclasses.Object.relative_filepath) which depends on the current working directory.

For example in `templates/python/material/function.html`:

```django
{% extends "_base/function.html" %}
{% block source scoped %}
  {% if config.show_source and function.source %}
    <details class="quote">
      <summary>{{ lang.t("Source code in") }} <code>{{ function.relative_package_filepath }}</code></summary>
      {{ function.source|highlight(language="python", linestart=function.lineno, linenums=True) }}
    </details>
  {% endif %}
{% endblock source %}
```

I'm considering adding some logic to the templates to use the relative filepath if its not absolute, and fallback to the relative package filepath.

**Reply by pawamoy** [reply]
Will be available in next release.

**Reply by Andrew-S-Rosen** [reply]
Fantastic to hear. Thank you!!
